msal/src/main/java/com/microsoft/identity/client/AccountAdapter.java

```diff
@@ -98,8 +118,13 @@ public List<ICacheRecord> filter(@NonNull final List<ICacheRecord> records) {
 
         private boolean hasNoCorrespondingHomeAccount(@NonNull final ICacheRecord guestRecord,
                                                       @NonNull final List<ICacheRecord> homeRecords) {
+            final String methodTag = CacheRecordFilter.class.getSimpleName() + ":hasNoCorrespondingHomeAccount";
             // Init our sought value
             final String guestAccountHomeAccountId = guestRecord.getAccount().getHomeAccountId();
+            if (guestAccountHomeAccountId == null){
```

Comment:
Do you also want to check for empty string here?

Reply:
Addressed